plugins/modules/ipapermission.py

```diff
@@ -0,0 +1,492 @@
+                                           "perm_target_filter")
+    perm_target = module_params_get(ansible_module, "perm_target")
+    perm_target_to = module_params_get(ansible_module, "perm_target_to")
+    perm_target_from = module_params_get(ansible_module, "perm_target_from")
+    member_of = module_params_get(ansible_module, "member_of")
+    target_group = module_params_get(ansible_module, "target_group")
+    object_type = module_params_get(ansible_module, "object_type")
+    rename = module_params_get(ansible_module, "rename")
+    privilege = module_params_get(ansible_module, "privilege")
+    action = module_params_get(ansible_module, "action")
+
+    # state
+    state = module_params_get(ansible_module, "state")
+
+    # Check parameters
+
+    invalid = []
+
+    if state == "present":
+        if len(names) != 1:
+            ansible_module.fail_json(
+                msg="Only one permission can be added at a time.")
+        if action == "member":
+            invalid = ["privilege", "rename"]
+
+    if state == "renamed":
+        if len(names) != 1:
+            ansible_module.fail_json(
+                msg="Only one permission can be renamed at a time.")
+        if action == "member":
+            ansible_module.fail_json(
+                msg="Member Privileges cannot be renamed")
+        invalid = ["perm_rights", "attrs", "perm_bind_rule_type",
+                   "perm_location", "extra_target_filter",
+                   "perm_target_filter", "perm_target", "perm_target_to",
+                   "perm_target_from", "member_of", "target_group",
+                   "object_type", "no_members"]
+
+    if state == "absent":
+        if len(names) < 1:
+            ansible_module.fail_json(msg="No name given.")
+        invalid = ["perm_rights", "attrs", "perm_bind_rule_type",
+                   "perm_location", "extra_target_filter",
+                   "perm_target_filter", "perm_target", "perm_target_to",
+                   "perm_target_from", "member_of", "target_group",
+                   "object_type"]
+        if action == "permission":
+            invalid.append("privilege")
+
+    for x in invalid:
+        if vars()[x] is not None:
+            ansible_module.fail_json(
+                msg="Argument '%s' can not be used with action "
+                "'%s' and state '%s'" % (x, action, state))
+
+    # Init
+
+    changed = False
+    exit_args = {}
+    ccache_dir = None
+    ccache_name = None
+    try:
+        if not valid_creds(ansible_module, ipaadmin_principal):
+            ccache_dir, ccache_name = temp_kinit(ipaadmin_principal,
+                                                 ipaadmin_password)
+        api_connect()
+
+        commands = []
+        for name in names:
+            # Make sure permission exists
+            res_find = find_permission(ansible_module, name)
+
+            # Create command
+            if state == "present" or state == "rename":
```

Comment:
Please, use `renamed` for the state name. I also suggest that `present` and `renamed` states are handled separately, with the rename state only allowing to rename the permission.